Mesh4all/m4a-firmware
# Automatization: m4a-template-generator

### Contribution description

In this PR is created an `pip` package, that contains various pythons scripts that will be used to generate all templates required to work with modules + boards_support + test and examples. this implement an integration to the shell comand of the system locating the m4a-generator app in /usr/bin files. Resolved a simple makefile to generate the template of each cases.

#### installation of the m4a-generator
by an shell script
```
./dist/tools/utils/m4agen-installer.sh
```
by pip
```sh
cd dist/tools/m4a-generator
sudo pip install .
```



### Testing procedure
To test if the installation was succefully, It's recommended run: `m4agen --version` this is a command to check the version of m4a-generator.

in the original repository you could try testing tipying:
```sh
make generate-example
```

if you want to uninstall the m4agen run: 
```sh
pip uninstall m4agen
```




### Issues/PRs references
